File: src/Problem1/Solution.java
// Accepted 11/22/2024
package Problem1;

import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int firstNumber = 0;
        int secondNumber = 0;
        
        var map = new HashMap<Integer, Integer>();
        
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            
            if (map.containsKey(complement)) {
                firstNumber = map.get(complement);
                secondNumber = i;
                break;
            }
            
            map.put(nums[i], i);
        }
        
        return new int[]{firstNumber, secondNumber};
    }
}

File: src/Problem1/Solution.py
# Accepted 12/4/2024
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        first_number = 0
        second_number = 0
        
        numbers = {}
        
        for i in range(len(nums)):
            complement = target - nums[i]
            
            if complement in numbers:
                first_number = numbers[complement]
                second_number = i
                break
            
            numbers[nums[i]] = i
        
        return [first_number, second_number]
